include/swift/Remote/MetadataReader.h

```diff
@@ -389,6 +389,90 @@ class MetadataReader {
     ContextDescriptorCache.clear();
   }
 
+  void dumpAsFullMetadata(RemoteAddress MetadataAddress) {
```

Comment:
I think there might be a place for this sort of debugging code in the runtime, but it shouldn't be compiled in unless assertions are enabled, and it seems like there ought to be a cleaner way to do it. For example, you could define a 'struct MetadataFieldDescriptor { name, offset, size }'; and pass in a RemoteAddress + descriptor array into a generic dump method. Could you hold off on adding this for some future/follow-up patch?